_commands/tools/grep.md

```diff
@@ -3,22 +3,100 @@
 
 grep
 -------
-TODO: Add documentation for this command by submitting a pull request.
-<!-- one line explanation would go here -->
 
-<!-- minimal example -->
+`grep` is a program for searching text files for lines that match regular exprssions. It can be used for all sorts of pattern-matching and text-based query analysis.
+
+<!--more-->
+
 ~~~ bash
-*short* example of command usage and output
+$ grep "wolf" animals.txt
+// finds and prints all animals containing "wolf" (wolf spider, grey wolf, wolfhound, etc.)
+$ grep "^wolf" animals.txt
+// finds and prints  all animals beginning with "wolf" (wolf spider, wolfhound, etc.)
+$ grep "wolf$" animals.txt
+// finds and prints all animals ending with "wolf" (grey wolf, etc.)
+$ grep "^c.*$"
+// finds and prints all animals beginning with the letter "c"
+$ grep "do+"
+// finds and prints all animals containing a three-character sequence that begins with "do"
 ~~~
 
-<!--more-->
+By default, `grep` will print all the matching lines to standard output (the console) with the matching parts colored. This behavior can be modified by using one or more of the the command line options `grep` offers.
 
 ### Useful Options / Examples
 
-#### Example command
+#### -c, -&#45;count
+
+##### Using the `-c` or `--count` option overrides the default output behavior of `grep`. Instead of printing all of the matches to the console, it will print the number of lines that match. This is a useful option if, for example, you only want to know if there are any matches (count &lt;&gt; 0) but aren't as concerned with what those matches actually are.
```

Comment:
This shows up as `count <> 0`. Should it just be `count > 0`?

Reply:
Yeah, it should. Fixed this.